draft-ietf-tls-tls13.md

```diff
@@ -2430,8 +2432,8 @@ The "extension_data" field of this extension contains a
 
 client_shares
 : A list of offered KeyShareEntry values in descending order of client preference.
-  This vector MUST NOT be empty. Clients not providing a KeyShare MUST instead
-  omit this extension from the ClientHello.
+  Servers MUST NOT negotiate any connection if this vector is empty and instead
+  MUST respond with a HelloRetryRequest (see {{hello-retry-request}}).
```

Comment:
Even with that, there's still a conflict on how to handle a lack of mutually supported key share when there is a mutually supported group: retry or negotiate lower priority suite? This complication is an example of why I don't really like the retry mechanic.